### ISSUE TITLE
Document encoding of `formatQuery`

### ISSUE THREAD

**Issue by dustin** [haskellari/postgresql-simple#80]
`formatQuery` returns a `ByteString` which is meant to be printed out (debugging, logging, etc...) but the character encoding is not specified.  I assume it's UTF-8, but it'd be good to clarify that in the docs.  (Or alternatively, return `Text`).

```haskell
formatQuery :: ToRow q => Connection -> Query -> q -> IO ByteString
```

**Comment by phadej** [comment]
It's not that simple, it relies on what `libpq` does, e.g. in https://hackage.haskell.org/package/postgresql-libpq-0.9.4.3/docs/Database-PostgreSQL-LibPQ.html#v:escapeStringConn

**Comment by phadej** [comment]
There is already a paragraph saying

> This function is exposed to help with debugging and logging. Do not use it to prepare queries for execution.

And the resulting type is `ByteString`. Perfectly you wouldn't assume anything about it, i.e. your logger can take arbitrary `ByteString`s and log them in human readable way when possible and not when they are not.

**Comment by dustin** [comment]
Yeah, that's what I'm doing.  I just want to make sure it's even a sensible first guess as well as what might affect it.

This is one of those things that might work well in development but not work at all in production and then wouldn't be noticed until stuff was broken and we needed it.  :)

**Comment by phadej** [comment]
As I said, I'll avoid giving any general advice here. Figure out what works in your setup.

**Comment by pbrinkmeier** [comment]
@phadej At this point it might be a sensible option to `setClientEncoding "UTF8"` on *every* connection.

This causes the Postgres server to convert textual values to UTF-8 even if they are stored as another encoding before sending them to the client, which means that this works even if the server itself uses some different obscure format.

`UTF8` is a support character set in Postgres since version 8.1, which was released in 2010 (currently the latest supported version is 12). Which means the chance of breaking stuff by doing this is fairly low.

Arguably the codebase assumes that the server sends UTF-8 already, e.g. in the `instance FromField Text`, which throws a `ConversionError` if the given `ByteString` isn't valid UTF-8.

I am happy to provide a PR including tests!

Doing this would enable a `formatQuery'` implementation that returns `String` or `Text`.
